tutorials/plot_background_filtering.py

```diff
@@ -366,8 +382,120 @@ def plot_signal(x, offset):
+# of the filter:
+
+sos = signal.iirfilter(2, f_p / nyq, btype='low', ftype='butter', output='sos')
+plot_filter(sos, 'Butterworth order=2', freq, gain)
+
+# Eventually this will just be from scipy signal.sosfiltfilt, but 0.18 is
+# not widely adopted yet (as of June 2016), so we use our wrapper...
+sosfiltfilt = mne.fixes.get_sosfiltfilt()
+x_shallow = sosfiltfilt(sos, x)
+
+###############################################################################
+# The falloff of this filter is not very steep.
+#
+# .. warning:: For brevity, we do not show the phase of these filters here.
+#              In the FIR case, we can design linear-phase filters, and
+#              compensate for the delay if necessary. This cannot be done
+#              with IIR filters, and as the filter order increases, the
+#              phase distortion near and in the transition band worsens.
+#              However, if acausal (forward-backward) filtering can be used,
+#              e.g. with :func:`scipy.signal.filtfilt`, these phase issues
+#              can be mitigated.
+#
+# .. note:: Here we have made use of second-order sections (SOS)
+#           by using :func:`scipy.signal.sosfilt` and, under the
+#           hood, :func:`scipy.signal.zpk2sos` when passing the
+#           ``output='sos'`` keyword argument to
+#           :func:`scipy.signal.iirfilter`. The filter definitions
+#           given in :ref:`filtering-basics` use the polynomial
+#           numerator/denominator (sometimes called "tf") form ``(b, a)``,
+#           which are theoretically equivalent to the SOS form used here.
+#           In practice, however, the SOS form can give much better results
+#           due to issues with numerical precision (see
+#           :func:`scipy.signal.sosfilt` for an example), so SOS should be
+#           used when possible to do IIR filtering.
+#
+# Let's increase the order, and note that now we have better attenuation,
+# with a longer impulse response:
+
+sos = signal.iirfilter(8, f_p / nyq, btype='low', ftype='butter', output='sos')
+plot_filter(sos, 'Butterworth order=8', freq, gain)
+x_steep = sosfiltfilt(sos, x)
+
+###############################################################################
+# There are other types of IIR filters that we can use. For a complete list,
+# check out the documentation for :func:`scipy.signal.iirdesign`. Let's
+# try a Chebychev (type I) filter, which trades off ripple in the pass-band
+# to get better attenuation in the stop-band:
+
+sos = signal.iirfilter(8, f_p / nyq, btype='low', ftype='cheby1', output='sos',
+                       rp=1)  # dB of acceptable pass-band ripple
+plot_filter(sos, 'Chebychev-1 order=8, ripple=1 dB', freq, gain)
+
+###############################################################################
+# And if we can live with even more ripple, we can get it slightly steeper,
+# but the impulse response begins to ring substantially longer:
```

Comment:
Maybe for slow people like me, it might make sense to point out the different x axes compared to the previous figure.